extensions/ql-vscode/test/unit-tests/databases/db-manager.test.ts

```diff
@@ -0,0 +1,227 @@
+import { ensureDir, readJSON, remove, writeJson } from "fs-extra";
+import { join } from "path";
+import { DbConfig } from "../../../src/databases/config/db-config";
+import { DbConfigStore } from "../../../src/databases/config/db-config-store";
+import {
+  flattenDbItems,
+  isLocalDatabaseDbItem,
+  isLocalListDbItem,
+  isRemoteUserDefinedListDbItem,
+  LocalDatabaseDbItem,
+  LocalListDbItem,
+  RemoteUserDefinedListDbItem,
+} from "../../../src/databases/db-item";
+import { DbManager } from "../../../src/databases/db-manager";
+import {
+  createDbConfig,
+  createLocalDbConfigItem,
+} from "../../factories/db-config-factories";
+import { createMockApp } from "../../__mocks__/appMock";
+
+// Note: Although these are "unit tests" (i.e. not integrating with VS Code), they do
+// test the interaction/"integration" between the DbManager and the DbConfigStore.
+describe("db manager", () => {
+  const extensionPath = join(__dirname, "../../..");
+  const tempWorkspaceStoragePath = join(__dirname, "test-workspace");
+  const app = createMockApp({
+    extensionPath,
+    workspaceStoragePath: tempWorkspaceStoragePath,
+  });
+  const dbConfigStore = new DbConfigStore(app);
+  const dbManager = new DbManager(app, dbConfigStore);
+
+  beforeEach(async () => {
+    await ensureDir(tempWorkspaceStoragePath);
+  });
+
+  afterEach(async () => {
+    await remove(tempWorkspaceStoragePath);
+  });
+
+  const dbConfigFilePath = join(
+    tempWorkspaceStoragePath,
+    "workspace-databases.json",
+  );
+
+  describe("renaming items", () => {
+    const remoteList = {
+      name: "my-list-1",
+      repositories: ["owner1/repo1", "owner1/repo2"],
+    };
+    const localDb = createLocalDbConfigItem({ name: "db1" });
+    const localList = {
+      name: "my-list-1",
+      databases: [localDb],
+    };
+
+    it("should rename remote db list", async () => {
+      const dbConfig = createDbConfig({
+        remoteLists: [remoteList],
+        localLists: [localList],
+      });
+
+      await saveDbConfig(dbConfig);
+
+      const dbItem = getRemoteUserDefinedListDbItem("my-list-1");
+
+      await dbManager.renameList(dbItem, "my-list-2");
+
+      const dbConfigFileContents = await readDbConfigDirectly();
+
+      // Check that the remote list has been renamed
+      const remoteLists = dbConfigFileContents.databases.remote.repositoryLists;
+      expect(remoteLists.length).toBe(1);
+      expect(remoteLists[0]).toEqual({
+        name: "my-list-2",
+        repositories: ["owner1/repo1", "owner1/repo2"],
+      });
+
+      // Check that the local list has not been renamed
+      const localLists = dbConfigFileContents.databases.local.lists;
+      expect(localLists.length).toBe(1);
+      expect(localLists[0]).toEqual({
+        name: "my-list-1",
+        databases: [localDb],
+      });
+    });
```

Comment:
We should also do `dbConfigStore.dispose()` at the end of each test (ideally in a `afterEach()`, with `beforeEach` creating the objects).